src/openrct2/scripting/ScEntity.hpp

```diff
@@ -1188,6 +1190,39 @@ namespace OpenRCT2::Scripting
                 peep->StaffOrders = value;
             }
         }
+
+        bool has_patrolarea() const
+        {
+            auto staff = GetStaff();
+            return staff->HasPatrolArea();
+        }
+
+        std::vector<DukValue> getPatrolArea()
+        {
+            std::vector<DukValue> patrolArea;
+            auto ctx = GetContext()->GetScriptEngine().GetContext();
+            auto staff = GetStaff();
+            if (staff == nullptr)
+            {
+                return {};
+            }
+            for (int32_t y = 0; y < 64; y++)
+            {
+                auto y_index = y << 7;
```

Comment:
don't use shifts unless you absolutely have to. Leave it to the compiler to optimise them.

Reply:
Also you could have just done +128 instead of ++ to avoid the multiplication.

Reply:
I was simply copying the shift logic used in the Staff class for consistency, but there's myriad ways to write this code of course. The most "readable" implementation, I think, would be keeping the single ++'s and using expanded multiplication like 
`y_index = y * 32 * 4;`

Reply:
The 32 is for the number of big x/y coordinates in 1 small coordinate. We have a constant for that: `COORDS_XY_STEP`. 

The 4 is for the number of tiles in a single piece of patrol area. I don't know if we have a constant for it, but if we haven't, please introduce one.